### DISCUSSION TITLE
Use custom definition for a pre-defined languageID (YAML)

### DISCUSSION THREAD

**Discussion by GSri30** [TypeFox/monaco-languageclient#507] [Q&A]
So I am trying to use `yaml` as LanguageID. I can see (from `monaco.languages.getLanguages()`) that there are some pre-defined languages that are already loaded into monaco. This list also contains `yaml`.

Now I have my DSL built upon `yaml` (I have specific keys and values domain). When I'm trying to integrate my react app with my language server, I'm not getting few functionalities (eg. `textDocument/completion`). I logged the messages on the server and I wasn't getting `textDocument/completion` method being triggered from client side (Though few other methods like `textDocument/didChange`, `textDocument/publishDiagnostics`, `textDocument/foldingRange`) were getting triggered.

What could be the fix.

**Comment by GSri30** [comment]
> So I am trying to use `yaml` as LanguageID. I can see (from `monaco.languages.getLanguages()`) that there are some pre-defined languages that are already loaded into monaco. This list also contains `yaml`.
> 
> Now I have my DSL built upon `yaml` (I have specific keys and values domain). When I'm trying to integrate my react app with my language server, I'm not getting few functionalities (eg. `textDocument/completion`). I logged the messages on the server and I wasn't getting `textDocument/completion` method being triggered from client side (Though few other methods like `textDocument/didChange`, `textDocument/publishDiagnostics`, `textDocument/foldingRange`) were getting triggered.
> 
> What could be the fix.

CC: @CGNonofr @kaisalmen

**Reply by CGNonofr** [reply]
Does the language server declare the capability ?

**Reply by GSri30** [reply]
Yeah. It does declare the capability. If I try to use the same language server and language client with different languageID config, its working. Its just that, with `yaml` as LanguageID, its not.

**Reply by CGNonofr** [reply]
Can you provide a minimal repo reproducing the issue?
